### PR TITLE
added message sign and verify functionality

### PR DESCRIPTION
**Pull request by paolocappelletti**

### CODE REVIEW

File: src/message.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    compressed: !!(flagByte & 12),
+    recovery: flagByte & 3,
+    signature: buffer.slice(1)
+  }
+}
+
+function _magicHash(message) {
+  const MAGIC_BYTES = new Buffer('Zcash Signed Message:\n');
+  var prefix1 = varuint.encode(MAGIC_BYTES.length);
+  var messageBuffer = new Buffer(message);
```
Comment (cronicc) [inline src/message.js]: (node:5454) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.

Comment (paolocappelletti) [reply src/message.js]: fixed!

**Comment by cronicc** [comment]
Force pushed signing the last commit.

**Comment by Ukyithar** [comment]
_magicHash

**Comment by Ukyithar** [comment]
https://github.com/HorizenOfficial/zencashjs/pull/32#issuecomment-702339877

**Pull request merged**